habit/views.py

```diff
@@ -0,0 +1,65 @@
+from rest_framework.generics import CreateAPIView, DestroyAPIView, ListAPIView, RetrieveAPIView, UpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+
+from habit.models import Habit
+from habit.paginators import PageNumberPagination
+from habit.serializers import HabitSerializer
+from users.permissions import IsUser
+
+
+class HabitCreateAPIView(CreateAPIView):
+    queryset = Habit.objects.all()
+    serializer_class = HabitSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def perform_create(self, serializer):
+        habit = serializer.save()
+        habit.user = self.request.user
+        habit.save()
+
+
+class PublicHabitListView(ListAPIView):
+    queryset = Habit.objects.all()
+    serializer_class = HabitSerializer
+    pagination_class = PageNumberPagination
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        return Habit.objects.filter(is_public=True)
```

Comment:
Если переопределяешь метод - используй лучше super() - меньше вероятности совершить ошибку
```suggestion
        return super().get_queryset().filter(is_public=True)
```
Ниже тоже